Tests/test_MarkovModel.py

```diff
@@ -234,6 +242,254 @@ def test_topcoder5(self):
+        markov_model_load = MarkovModel.load(handle)
+
+        self.assertEqual(''.join(markov_model_load.states), states)
+        self.assertEqual(''.join(markov_model_load.alphabet), alphabet)
+        self.assertTrue(array_equal(markov_model_load.p_initial, p_initial))
+        self.assertTrue(array_equal
+                        (markov_model_load.p_transition, p_transition))
+        self.assertTrue(array_equal(markov_model_load.p_emission, p_emission))
+
+    def test_train_bw(self):
+        random.seed(0)
+        states = ["0", "1", "2", "3"]
+        alphabet = ["A", "C", "G", "T"]
+        training_data = ["AACCCGGGTTTTTTT", "ACCGTTTTTTT",
+                         "ACGGGTTTTTT", "ACCGTTTTTTTT"]
+
+        output_p_initial = array([0.2275677, 0.29655611,
+                                  0.24993822, 0.22593797])
+        output_p_transition = array(
+            [[5.16919807e-001, 3.65825814e-033, 4.83080193e-001, 9.23220689e-042],
+             [3.65130247e-001,
+              1.00000000e-300,
+              6.34869753e-001,
+              1.00000000e-300],
+             [8.68776164e-001,
+              1.02254304e-034,
+              1.31223836e-001,
+              6.21835051e-047],
+             [3.33333333e-301, 3.33333333e-001, 3.33333333e-301, 6.66666667e-001]])
+        output_p_emission = array(
+            [[2.02593570e-301, 2.02593570e-301, 2.02593570e-301, 1.00000000e+000],
+             [1.00000000e-300,
+              1.00000000e-300,
+              1.00000000e+000,
+              1.09629016e-259],
+             [3.26369779e-301,
+              3.26369779e-301,
+              3.26369779e-301,
+              1.00000000e+000],
+             [3.33333333e-001, 6.66666667e-001, 3.33333333e-301, 3.33333333e-301]])
+
+        markov_model = MarkovModel.train_bw(states, alphabet, training_data)
+        self.assertEqual(''.join(markov_model.states), ''.join(states))
+        self.assertEqual(''.join(markov_model.alphabet), ''.join(alphabet))
+        self.assertTrue(array_equal(
+            around(markov_model.p_initial, decimals=3),
+            around(output_p_initial, decimals=3)))
+        self.assertTrue(array_equal(around(
+            markov_model.p_transition, decimals=3),
+            around(output_p_transition, decimals=3)))
+        self.assertTrue(
+            array_equal(around(
+                        markov_model.p_emission, decimals=3),
+                        around(output_p_emission, decimals=3)))
+
+    def test_forward(self):
+        states = ["CP", "IP"]
+        outputs = [2, 1, 0]
+        lp_initial = log([1.0, 0.0000001])
+        lp_transition = log([[0.7, 0.3], [0.5, 0.5]])
+        lp_emission = log([[0.6, 0.1, 0.3], [0.1, 0.7, 0.2]])
+
+        matrix = array([[0., -1.5606477, -3.07477539, -3.84932984],
+                        [-16.11809565, -2.4079455, -3.27544608, -4.5847794]])
+        self.assertTrue(
+            array_equal(around(MarkovModel._forward(len(states),
```

Comment:
As above, put the ``array_equal`` on the first line so that the two things being compared is clearer.